packages/algob/src/lib/script-checkpoints.ts

```diff
@@ -122,17 +123,17 @@ export class CheckpointRepoImpl implements CheckpointRepo {
     return undefined;
   }
 
-  registerASA (networkName: string, name: string, creator: string): CheckpointRepo {
-    registerASA(this._ensureNet(this.precedingCP, networkName), name, creator);
-    registerASA(this._ensureNet(this.strippedCP, networkName), name, creator);
-    registerASA(this._ensureNet(this.allCPs, networkName), name, creator);
+  registerASA (networkName: string, name: string, info: ASAInfo): CheckpointRepo {
+    registerASA(this._ensureNet(this.precedingCP, networkName), name, info);
+    registerASA(this._ensureNet(this.strippedCP, networkName), name, info);
+    registerASA(this._ensureNet(this.allCPs, networkName), name, info);
     return this;
   }
 
-  registerASC (networkName: string, name: string, creator: string): CheckpointRepo {
-    registerASC(this._ensureNet(this.precedingCP, networkName), name, creator);
-    registerASC(this._ensureNet(this.strippedCP, networkName), name, creator);
-    registerASC(this._ensureNet(this.allCPs, networkName), name, creator);
+  registerASC (networkName: string, name: string, info: ASCInfo): CheckpointRepo {
+    registerASC(this._ensureNet(this.precedingCP, networkName), name, info);
+    registerASC(this._ensureNet(this.strippedCP, networkName), name, info);
+    registerASC(this._ensureNet(this.allCPs, networkName), name, info);
```

Comment:
I added some doc text to dropbox doc:

Existing checkpoint state loading

Checkpoints are saved as files in  `artifacts/scripts/`. Their names are derived from corresponding script names by adding `.cp.yaml`  to them.
Checkpoints are read during deploy and run tasks. They are used to load the previous states of the execution and this way the scripts can continue run with previously cached data. Persisted checkpoint files are only increments of the existing state and they mustn’t redefine values between them (i.e. any name clash that is introduced by hand should result in an error).
This is realized by having three checkpoint states during script execution runtime:

- `strippedCP` — Contains only what currently running script will save into it’s checkpoint
- `precedingCP` — Contains only what the currently running script can query (prevents state cycles)
- `allCPs` — Contains all asset definitions to prevent redefinition of what future scripts defined

All three views on the checkpoint states allow to control script execution so that they wouldn’t introduce different state that was already introduced with names that have ascii-larger (sort order) names.